test/FunctionalTests/Microsoft.OData.Core.Tests/AsyncYieldStream.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.OData.Tests
+{
+    /// <summary>
+    /// Stream wrapper that yields in every async method
+    /// to ensure the read/write operations complete asynchronously
+    /// even when the operate completely in-memory. This helps simulate
+    /// async I/O in unit tests without making requests out of the process.
+    /// </summary>
+    internal class AsyncYieldStream : Stream

```

Comment:
ODL test project doesn't contain similar class?

Reply:
@xuzhg With the `AsyncStream`, I was not able to repro the issue faced by the customer https://github.com/OData/AspNetCoreOData/issues/1320. I added class `AsyncYieldStream` with `await Task.Yield()` in the async methods to help repro the issue and fix it. 

`await Task.Yield()` ensures that method yields control back to the caller immediately, allowing other tasks to run without blocking the thread.